### PR TITLE
[PL-42276] Active PL Feature Flags

### PR DESCRIPTION
**Pull request by saifeemustafaqharness**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): https://harness.atlassian.net/browse/PL-42276
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/platform/references/active-platform-feature-flags.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+title: Active Feature Flags - Platform
+description: You can view all the feature flags curretly active and in production.
+sidebar_position: 11
+helpdocs_topic_id: //not sure what to put here
+helpdocs_category_id: //not sure what to put here
+helpdocs_is_private: false
+helpdocs_is_published: true
```
Comment (brian-f-harness) [inline docs/platform/references/active-platform-feature-flags.md:8]:
```suggestion
```

Comment (brian-f-harness) [reply docs/platform/references/active-platform-feature-flags.md:8]: @saifeemustafaqharness These are legacy fields you find in topics that were converted from the old docs system. They aren't necessary for new topics.

File: docs/platform/references/active-platform-feature-flags.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+---
+
+The list on this page indicates all the feature flags currently active. Contact [Harness Support](mailto:support@harness.io) to enable the features.
+
+### What are feature flags
+
+Feature flags are a way that developers can conditionally turn certain sections of their code (or code paths) on or off. You can think of feature flags as an extension or a way to put changes into production behind a flag and turn them on in a controlled way later (or hide and remove them in the same way). They form the basis of feature flag.
```
Comment (brian-f-harness) [inline docs/platform/references/active-platform-feature-flags.md:15]:
```suggestion
```

Comment (brian-f-harness) [reply docs/platform/references/active-platform-feature-flags.md:15]: You can use the verbiage CD uses in the What's supported topic.

https://developer.harness.io/docs/continuous-delivery/cd-integrations#active-cd-feature-flags

**Review by brian-f-harness — CHANGES_REQUESTED** [review]
@saifeemustafaqharness I suggest moving this content to the What's supported topic to mirror how CD presents the FF list. This was Sid's original request.

File: docs/platform/references/active-platform-feature-flags.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+| PL_UPDATE_EMAILS_VIA_SCIM                   |             |
+| PL_NO_EMAIL_FOR_SAML_ACCOUNT_INVITES        |             |
+| PL_LDAP_PARALLEL_GROUP_SYNC                 |             |
+| PL_NEW_SCIM_STANDARDS                       |             |
+| PL_USE_CREDENTIALS_FROM_DELEGATE_FOR_GCP_SM |             |
+| PL_DELEGATE_TASK_CAPACITY_CHECK             |             |
+| PL_FAVORITES                                |             |
+| PL_ALLOW_TO_SET_PUBLIC_ACCESS               |             |
+| PL_GCP_OIDC_AUTHENTICATION                  |             |
+| PL_CENTRAL_NOTIFICATIONS                    |             |
```
Comment (brian-f-harness) [inline docs/platform/references/active-platform-feature-flags.md:28]:
```suggestion
| `PL_UPDATE_EMAILS_VIA_SCIM`                |             |
| `PL_NO_EMAIL_FOR_SAML_ACCOUNT_INVITES`        |             |
| `PL_LDAP_PARALLEL_GROUP_SYNC`                 |             |
| `PL_NEW_SCIM_STANDARDS`                     |             |
| `PL_USE_CREDENTIALS_FROM_DELEGATE_FOR_GCP_SM` |             |
| `PL_DELEGATE_TASK_CAPACITY_CHECK`             |             |
| `PL_FAVORITES`                               |             |
| `PL_ALLOW_TO_SET_PUBLIC_ACCESS`               |             |
| `PL_GCP_OIDC_AUTHENTICATION`                  |             |
| `PL_CENTRAL_NOTIFICATION`                    |             |
```

File: docs/platform/platform-whats-supported.md [hunk 4]
```diff
@@ -224,3 +224,23 @@ Harness includes native TUF support via the following:
 - Enforce the rotation of secrets and key management practices:
   - Harness provides [token key rotation natively](/docs/platform/automation/api/add-and-manage-api-keys#rotate-tokens).
 - Continuous Verification: TUF recommends the verification of deployments akin to [Harness Continuous Verification](/docs/continuous-delivery/verify/verify-deployments-with-the-verify-step).
+
+## Active Platform Feature flags
+
+Some Harness Platform features are released behind feature flags to get feedback from specific customers before releasing the features to the general audience.
```
Comment (brian-f-harness) [inline docs/platform/platform-whats-supported.md:230]:
```suggestion
Some Harness Platform features are released behind feature flags to get feedback from specific customers before releasing the features to the general audience. Feature development statuses are categorized as [Beta, GA, or Limited GA](/docs/platform/get-started/release-status).
```

File: docs/platform/platform-whats-supported.md [hunk 5]
```diff
@@ -224,3 +224,23 @@ Harness includes native TUF support via the following:
 - Enforce the rotation of secrets and key management practices:
   - Harness provides [token key rotation natively](/docs/platform/automation/api/add-and-manage-api-keys#rotate-tokens).
 - Continuous Verification: TUF recommends the verification of deployments akin to [Harness Continuous Verification](/docs/continuous-delivery/verify/verify-deployments-with-the-verify-step).
+
+## Active Platform Feature flags
```
Comment (brian-f-harness) [inline docs/platform/platform-whats-supported.md:228]:
```suggestion
## Active Platform feature flags
```

File: docs/platform/platform-whats-supported.md [hunk 6]
```diff
@@ -224,3 +224,23 @@ Harness includes native TUF support via the following:
...
+
+The following table describes active feature flags relevant to Harness Platform.
+
+To enable a feature flag in your Harness account, contact [Harness Support](mailto:support@harness.io).
+
+| Flag                                          | Description |
+|-----------------------------------------------|-------------|
+| `PL_NO_EMAIL_FOR_SAML_ACCOUNT_INVITES`        | When activated, it prevents sending email invitations to users of accounts using SSO for authentication during onboarding. |
+| `PL_LDAP_PARALLEL_GROUP_SYNC`                 | Enable User Group sync operation to fetch data from the LDAP server in parallel. Only enable this if the LDAP server can handle the load. |
+| `PL_NEW_SCIM_STANDARDS`                       | Enabling the `PL_NEW_SCIM_STANDARDS` feature flag ensures compliance with SCIM 2.0 standards by including the meta fields `createdAt`, `lastUpdated`, `version`, and `resourceType in CRUD operation responses on users or user groups. |
```
Comment (brian-f-harness) [inline docs/platform/platform-whats-supported.md:240]:
```suggestion
| `PL_NEW_SCIM_STANDARDS`                       | Enabling the `PL_NEW_SCIM_STANDARDS` feature flag ensures compliance with SCIM 2.0 standards by including the meta fields `createdAt`, `lastUpdated`, `version`, and `resourceType` in CRUD operation responses on users or user groups. |
```

File: docs/platform/platform-whats-supported.md [hunk 7]
```diff
@@ -224,3 +224,23 @@ Harness includes native TUF support via the following:
...
+| `PL_USE_CREDENTIALS_FROM_DELEGATE_FOR_GCP_SM` | Enabling this Feature Flag will let you use delegate credentials to access the Google Cloud Platform Secret Manager. |
+| `PL_DELEGATE_TASK_CAPACITY_CHECK`             | When enabled, tasks of the accounts will be broadcasted until task timeout other wise the default behavior is that tasks will be broadcasted up to 3 times to delegates. |
```
Comment (brian-f-harness) [inline docs/platform/platform-whats-supported.md:242]:
```suggestion
| `PL_DELEGATE_TASK_CAPACITY_CHECK`             | When enabled, account tasks will be broadcasted until timeout. The default behavior is to broadcast tasks up to 3 times to delegates. |
```

File: docs/platform/platform-whats-supported.md [hunk 8]
```diff
@@ -224,3 +224,23 @@ Harness includes native TUF support via the following:
...
+| `PL_FAVORITES`                                | Allows you to set the frequently accessed projects and connectors as favorites. |
+| `PL_ALLOW_TO_SET_PUBLIC_ACCESS`               | Allows pipeline executions marked for public view to be accessed without authentication. You can share execution URLs, including console logs, without requiring users to sign in. |
+| `PL_GCP_OIDC_AUTHENTICATION`                  | Enabling this feature flag allows GCP connectors using OpenID Connect (OIDC) to let Harness communicate directly with GCP through OIDC. |
+| `PL_CENTRAL_NOTIFICATION`                     | Enable/disable Centralized Notifications Management. |
```
Comment (brian-f-harness) [inline docs/platform/platform-whats-supported.md:246]:
```suggestion
| `PL_CENTRAL_NOTIFICATION`                     | Enable/disable Centralized Notifications Management. For more information, go to [Manage delegate notifications](/docs/platform/notifications/manage-notifications). |
```

File: docs/platform/platform-whats-supported.md [hunk 9]
```diff
@@ -224,3 +224,23 @@ Harness includes native TUF support via the following:
...
+| `PL_NEW_SCIM_STANDARDS`                       | Enabling the `PL_NEW_SCIM_STANDARDS` feature flag ensures compliance with SCIM 2.0 standards by including the meta fields `createdAt`, `lastUpdated`, `version`, and `resourceType in CRUD operation responses on users or user groups. |
+| `PL_USE_CREDENTIALS_FROM_DELEGATE_FOR_GCP_SM` | Enabling this Feature Flag will let you use delegate credentials to access the Google Cloud Platform Secret Manager. |
+| `PL_DELEGATE_TASK_CAPACITY_CHECK`             | When enabled, tasks of the accounts will be broadcasted until task timeout other wise the default behavior is that tasks will be broadcasted up to 3 times to delegates. |
+| `PL_FAVORITES`                                | Allows you to set the frequently accessed projects and connectors as favorites. |
+| `PL_ALLOW_TO_SET_PUBLIC_ACCESS`               | Allows pipeline executions marked for public view to be accessed without authentication. You can share execution URLs, including console logs, without requiring users to sign in. |
```
Comment (brian-f-harness) [inline docs/platform/platform-whats-supported.md:244]:
```suggestion
| `PL_ALLOW_TO_SET_PUBLIC_ACCESS`               | Allows pipeline executions marked for access to public view without authentication. You can share execution URLs, including console logs, without requiring users to sign in. For more information, go to [Allow public access to pipeline executions](/docs/platform/pipelines/executions-and-logs/allow-public-access-to-executions/).|
```

File: docs/platform/platform-whats-supported.md [hunk 10]
```diff
@@ -224,3 +224,23 @@ Harness includes native TUF support via the following:
...
+| `PL_NO_EMAIL_FOR_SAML_ACCOUNT_INVITES`        | When activated, it prevents sending email invitations to users of accounts using SSO for authentication during onboarding. |
+| `PL_LDAP_PARALLEL_GROUP_SYNC`                 | Enable User Group sync operation to fetch data from the LDAP server in parallel. Only enable this if the LDAP server can handle the load. |
+| `PL_NEW_SCIM_STANDARDS`                       | Enabling the `PL_NEW_SCIM_STANDARDS` feature flag ensures compliance with SCIM 2.0 standards by including the meta fields `createdAt`, `lastUpdated`, `version`, and `resourceType in CRUD operation responses on users or user groups. |
+| `PL_USE_CREDENTIALS_FROM_DELEGATE_FOR_GCP_SM` | Enabling this Feature Flag will let you use delegate credentials to access the Google Cloud Platform Secret Manager. |
+| `PL_DELEGATE_TASK_CAPACITY_CHECK`             | When enabled, tasks of the accounts will be broadcasted until task timeout other wise the default behavior is that tasks will be broadcasted up to 3 times to delegates. |
+| `PL_FAVORITES`                                | Allows you to set the frequently accessed projects and connectors as favorites. |
```
Comment (brian-f-harness) [inline docs/platform/platform-whats-supported.md:243]:
```suggestion
| `PL_FAVORITES`                                | Allows you to set the frequently accessed projects and connectors as favorites. For more information, go to [Set favorites](/docs/platform/favorites/set-favorites/). |
```

File: docs/platform/platform-whats-supported.md [hunk 11]
```diff
@@ -224,3 +224,23 @@ Harness includes native TUF support via the following:
...
+| `PL_ALLOW_TO_SET_PUBLIC_ACCESS`               | Allows pipeline executions marked for public view to be accessed without authentication. You can share execution URLs, including console logs, without requiring users to sign in. |
+| `PL_GCP_OIDC_AUTHENTICATION`                  | Enabling this feature flag allows GCP connectors using OpenID Connect (OIDC) to let Harness communicate directly with GCP through OIDC. |
```
Comment (brian-f-harness) [inline docs/platform/platform-whats-supported.md:245]:
```suggestion
| `PL_GCP_OIDC_AUTHENTICATION`                  | Enabling this feature flag allows GCP connectors using OpenID Connect (OIDC) to let Harness communicate directly with GCP through OIDC. For more information, go to [Google Cloud Platform (GCP) connector settings reference](/docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference/). |
```

**Review by batraprashant — APPROVED** [review]

**Pull request merged**